fe/fe-core/src/main/java/org/apache/doris/analysis/CastExpr.java

```diff
@@ -269,15 +270,28 @@ public void analyze() throws AnalysisException {
 
         // this cast may result in loss of precision, but the user requested it
         if (childType.matchesType(type)) {
-            // For types which has precision and scale, we also need to check quality between precisions and scales
-            if (!PrimitiveType.typeWithPrecision.contains(
-                    type.getPrimitiveType()) || ((((ScalarType) type).decimalPrecision()
-                    == ((ScalarType) childType).decimalPrecision()) && (((ScalarType) type).decimalScale()
-                    == ((ScalarType) childType).decimalScale()))) {
+            if (PrimitiveType.typeWithPrecision.contains(type.getPrimitiveType())) {
```

Comment:
Are you sure that the 3 "if" branches all set the 'noOp = true' ?

Reply:
for example, 
if type.isArrayType() == true, 
but 
((ArrayType) type).getContainsNull() != ((ArrayType) childType).getContainsNull(),
the noOp is remain set to false.